Sources/ComposableArchitecture/Internal/MainActorAssumeIsolated.swift

```diff
@@ -0,0 +1,8 @@
+@MainActor(unsafe)
+func assumeMainActorIsolated<T>(
+  _ operation: @MainActor () throws -> T,
+  file: StaticString = #fileID,
+  line: UInt = #line
+) rethrows -> T {
+  try operation()
+}
```

Comment:
The `Actor.assumeIsolated` function is supposed to be back deployed to iOS 13, and that does seem to be the case locally, but CI says iOS 17+. If we can figure out CI we can get rid of this.

Reply:
It was back deployed only in the Xcode 15.3 toolchain with Swift 5.10 AFAICT, not in 5.9 yet.

Reply:
Ahhh ok, so I can work around that with some `#if`s. Thanks @pyrtsa!